src/suites/cts/validation/createBindGroupLayout.spec.ts

```diff
@@ -111,17 +107,121 @@ g.test('dynamic set to true is allowed only for buffers', async t => {
   const success = bindingTypeInfo[type].type === 'buffer';
 
   const descriptor = {
-    bindings: [
-      {
-        binding: 0,
-        visibility: GPUShaderStage.FRAGMENT,
-        type,
-        hasDynamicOffset: true,
-      },
-    ],
+    bindings: [{ binding: 0, visibility: GPUShaderStage.FRAGMENT, type, hasDynamicOffset: true }],
   };
 
   t.expectValidationError(() => {
     t.device.createBindGroupLayout(descriptor);
   }, !success);
 }).params(poptions('type', bindingTypes));
+
+{
+  function pickExtraBindingTypes(type: GPUBindingType, extraTypeSame: boolean): GPUBindingType[] {
```

Comment:
nit: Please add comments about what this function is doing and why it helps us in the test.

From what I understand, we want to test cases when an additional binding pushes us over a limit, and when an additional binding does not push us over (`extraTypeSame`). If it's a storage buffer, then either types of storage buffer count toward the limit, ..., etc.